src/main/java/uk/gov/ons/ctp/response/action/service/ActionProcessingService.java

```diff
@@ -128,66 +176,55 @@ private boolean isBusinessNotification(ActionRequestContext context) {
     return actionRequest;
   }
 
   /** Deal with a single action cancel - the transaction boundary is here */
   @Transactional(propagation = Propagation.REQUIRES_NEW)
-  public void processActionCancel(final UUID actionId) {
-    Action action = actionRepo.findById(actionId);
+  public void processActionCancel(final Action action) {
 
     log.with("action_id", action.getId())
         .with("case_id", action.getCaseId())
         .with("action_plan_pk", action.getActionPlanFK())
         .info("processing action cancel");
 
-    transitionAction(action, ActionDTO.ActionEvent.CANCELLATION_DISTRIBUTED);
-
-    actionInstructionPublisher.sendActionInstruction(
-        action.getActionType().getHandler(), prepareActionCancel(action));
-  }
-
-  /**
-   * Take an action and using it, fetch further info from Case service in a number of rest calls, in
-   * order to create the ActionRequest
-   *
-   * @param action It all starts wih the Action
-   * @return The ActionRequest created from the Action and the other info from CaseSvc
-   */
-  private ActionCancel prepareActionCancel(final Action action) {
-    log.with("action_id", action.getId())
-        .with("case_id", action.getCaseId())
-        .with("action_plan_pk", action.getActionPlanFK())
-        .debug("Building ActionCancel to publish to downstream handler");
-    final ActionCancel actionCancel = new ActionCancel();
-    actionCancel.setActionId(action.getId().toString());
-    actionCancel.setResponseRequired(true);
-    actionCancel.setReason(CANCELLATION_REASON);
-    return actionCancel;
+    try {
+      actionStateService.transitionAction(action, ActionDTO.ActionEvent.CANCELLATION_DISTRIBUTED);
+    } catch (CTPException ctpExeption) {
+      throw new IllegalStateException(ctpExeption);
+    }
   }
 
   /**
    * Change the action status in db to indicate we have sent this action downstream, and clear
    * previous situation (in the scenario where the action has prev. failed)
    *
    * @param action the action to change and persist
-   * @param event the event to transition the action with
+   * @param actionType the action type
    * @throws CTPException if action state transition error
    */
-  private void transitionAction(final Action action, final ActionDTO.ActionEvent event) {
-    ActionDTO.ActionState nextState = null;
+  private void transitionAction(final Action action, final ActionType actionType) {
+
+    ActionDTO.ActionEvent event = getActionEvent(actionType);
 
     try {
-      nextState = actionSvcStateTransitionManager.transition(action.getState(), event);
+      actionStateService.transitionAction(action, event);
     } catch (CTPException ctpExeption) {
       throw new IllegalStateException(ctpExeption);
     }
+  }
 
-    action.setState(nextState);
-    action.setSituation(null);
-    action.setUpdatedDateTime(DateTimeUtil.nowUTC());
-    actionRepo.saveAndFlush(action);
+  private ActionDTO.ActionEvent getActionEvent(ActionType actionType) {
```

Comment:
Is there any reason we haven't imported `ActionEvent`? I don't see a conflict

Reply:
Yes they're internal public enums in the ActionDTO class so can't be imported directly, it's horrible but didn't want to make the change bigger.